python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -1050,19 +1050,18 @@ def reset_index(self, drop=False):
         return out.set_index(RangeIndex(len(self)))
 
     def take(self, positions, ignore_index=False):
-        positions = columnops.as_column(positions).astype("int32").data.mem
         out = DataFrame()
-        cols = [s._column for s in self._cols.values()]
-
-        result_cols = cpp_copying.apply_gather(cols, positions)
-
-        for i, col_name in enumerate(self._cols):
-            out[col_name] = result_cols[i]
-
-        if ignore_index:
-            out.index = RangeIndex(len(out))
-        else:
-            out.index = self.index.take(positions)
+        if self._cols:
+            positions = columnops.as_column(positions).astype("int32").data.mem
+            cols = [s._column for s in self._cols.values()]
+            result_cols = cpp_copying.apply_gather(cols, positions)
+            for i, col_name in enumerate(self._cols):
+                out[col_name] = result_cols[i]
+
+            if ignore_index:
+                out.index = RangeIndex(len(out))
+            else:
+                out.index = self.index.take(positions)
         return out
```

Comment:
You might want to return `self` if `not self._cols:` instead of a new empty dataframe - then `take` will return similar metadata from an empty dataframe, instead of losing it. I ask primarily as the MultiIndex implementer, which is metadata that typically we'd want to stick around.

Reply:
Would it be okay to return a new `DataFrame` in some situations, but `self` in others though (eg. is there an expectation that the returned DF is safe to change w/out modifying the original)?

Reply:
I believe we should return by copy regardless here and if/when we support inplace in the future that could return self instead.

Reply:
Indeed, its better to return a copy and if it has other issues like with MultiIndex we can deal with that as it comes.